File: Backtracking/allpossiblesubseqString/allpossiblesubseqString.py
def all_possible_subseq(s, current="", index=0):
    # base condition
    if index == len(s):
        print(current)
        return

    # include current character
    all_possible_subseq(s, current + s[index], index + 1)

    # exclude current character
    all_possible_subseq(s, current, index + 1)


# main
if __name__ == "__main__":
    s = input("Enter the string: ")
    print("The possible substrings are:")
    all_possible_subseq(s)

File: Backtracking/allpossiblesubseqString/allpossiblesubseqString.go
package main

import (
	"fmt"
)

func allPossibleSubseq(s string, current string, index int) {
	// base condition
	if index == len(s) {
		fmt.Println(current)
		return
	}

	// include current character
	allPossibleSubseq(s, current+string(s[index]), index+1)

	// exclude current character
	allPossibleSubseq(s, current, index+1)
}

func main() {
	var input string
	fmt.Println("Enter the string:")
	fmt.Scanln(&input)

	fmt.Println("The possible substrings are:")
	allPossibleSubseq(input, "", 0)
}
